01_Introduction/Hello/HelloWinForms/Form1.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Windows.Forms;
+
+namespace HelloWinForms
+{
+    public partial class Form1 : Form
+    {
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void button1_Click(object sender, EventArgs e)
+        {
+            MessageBox.Show($"Hello, {textBox1.Text}!", "Hello", MessageBoxButtons.OK);
+        }
+
+        private void textBox1_TextChanged(object sender, EventArgs e)
+        {
+            if (string.IsNullOrEmpty(textBox1.Text))
+                button1.Enabled = false;
+            else
+            {
+                button1.Enabled = true;
+            }
```

Comment:
This `if`-`else` can be simplified to a single line:
```
button1.Enabled = !string.IsNullOrEmpty(textBox1.Text);
```
Remember this, when you have the same thing in `if` and `else`, where the only difference is `true` and `false`;

Reply:
Great, thank you for the advice. I used it.